lib/session/sessionManager.ts

```diff
@@ -62,13 +72,130 @@ export class SessionManager {
     this.isManagingSessions = true;
     if (!options) options = {};
     if (options.maxConcurrentSessions) this.maxConcurrentSessions = options.maxConcurrentSessions;
+    // We are explicitly configuring the messageSession to timeout in 60 seconds if no new messages
+    // are received.
+    if (!options.maxMessageWaitTimeoutInSeconds) {
+      options.maxMessageWaitTimeoutInSeconds = Constants.defaultOperationTimeoutInSeconds;
+    }
+    this._maxConcurrentSessionsSemaphore = new Semaphore(this.maxConcurrenSessions);
+    this._maxPendingAcceptSessionsSemaphore = new Semaphore(
+      this.maxConcurrentAcceptSessionRequests
+    );
 
     for (let i = 0; i < this._maxConcurrentAcceptSessionRequests; i++) {
+      this._acceptSessionAndReceiveMessages(onSessionMessage, onError, options).catch((err) => {
+        log.error(err);
+      });
+    }
+  }
+
+  /**
+   * Close the session manager.
+   */
+  close(): void {
+    this._isCancelRequested = true;
+    this.isManagingSessions = false;
```

Comment:
What is the use case for `isManagingSessions`? It isnt getting used internally. It isnt exposed to the user either.

Reply:
It is being used in QueueClient.receiveMessgesFromSessions().
The method throws an error if sessionManager is already managingSessions. Thereby restricting the successive calls to receiveMessageFromSessions().

Actually I was thinking to move this check to SessionManager.manageSessions(). Thus we wouldn't need to repeat the same check for subscriptions in SubscriptionClient.

Reply:
And we can make it private.